validator/sawtooth_validator/journal/publisher.py

```diff
@@ -370,7 +370,8 @@ def _finalize_block(self, block):
                 else:
                     block.add_batch(batch)
                     self._committed_txn_cache.add_batch(batch)
-                state_hash = result.state_hash
+                if result.state_hash is not None:
+                    state_hash = result.state_hash
```

Comment:
This feels like a very awkward interface, that we have to fish through the batch results and look for a valid state hash.  It feels that there should be a property set on the scheduler once  self._scheduler.complete(block=True) returns that is the root_state_hash that was generated for all the batches validated.